iceoryx2-ffi/cxx/include/iox2/service_builder_publish_subscribe.hpp

```diff
@@ -149,29 +142,73 @@ inline ServiceBuilderPublishSubscribe<Payload, UserHeader, S>::ServiceBuilderPub
 
 template <typename Payload, typename UserHeader, ServiceType S>
 template <typename PayloadType>
-inline auto ServiceBuilderPublishSubscribe<Payload, UserHeader, S>::get_payload_type_name() ->
-    typename std::enable_if_t<HasPayloadTypeNameMember<PayloadType>::value, const char*> {
+inline auto ServiceBuilderPublishSubscribe<Payload, UserHeader, S>::get_payload_type_name()
+    -> internal::FromCustomizedPayloadTypeName<PayloadType> {
     return PayloadType::PAYLOAD_TYPE_NAME;
 }
 
 template <typename Payload, typename UserHeader, ServiceType S>
 template <typename PayloadType>
-inline auto ServiceBuilderPublishSubscribe<Payload, UserHeader, S>::get_payload_type_name() ->
-    typename std::enable_if_t<!HasPayloadTypeNameMember<PayloadType>::value, const char*> {
+inline auto ServiceBuilderPublishSubscribe<Payload, UserHeader, S>::get_payload_type_name()
+    -> internal::FromNonSlice<PayloadType> {
+    if (std::is_same_v<PayloadType, uint8_t>) {
+        return "u8";
+    }
+    if (std::is_same_v<PayloadType, uint16_t>) {
+        return "u16";
+    }
+    if (std::is_same_v<PayloadType, uint32_t>) {
+        return "u32";
+    }
+    if (std::is_same_v<PayloadType, uint64_t>) {
+        return "u64";
+    }
+    if (std::is_same_v<PayloadType, int8_t>) {
+        return "i8";
+    }
+    if (std::is_same_v<PayloadType, int16_t>) {
+        return "i16";
+    }
+    if (std::is_same_v<PayloadType, int32_t>) {
+        return "i32";
+    }
+    if (std::is_same_v<PayloadType, int64_t>) {
+        return "i64";
+    }
+    if (std::is_same_v<PayloadType, bool>) {
+        return "bool";
+    }
```

Comment:
Did not really review but just skimmed over since I was curious about the implementation and I leave it to elfenpiff to do the review :)

This is a good start but just as food for thought. Would it make sense to also add `float` and `double` to the list. They should be compatible to `f32` and `f64`. The `long double` is highly implementation dependent and can be between 96 and 128 bits.
Same with `char`, `signed char` and `unsigned char`. When it comes to transfer strings, a `[u8]` would be a natural choice in Rust but in C++ it would be a slice of `char`. The issue with `char` is that it is not even guaranteed to be 8 bit. So maybe it is a good idea to not have them as part of this types.

Reply:
I added float and double, but decided against char because of the missing size guarantees.